### DISCUSSION TITLE
common_parse_uint: return an index for the first unparsable character?

### DISCUSSION THREAD

**Discussion by ghost** [vlang/v#11719] [Ideas]
Maybe the result should be (u64, int), the first value indicating the parsed value, the second one the position of the first unparsable character index (or for simplicity -1 if everything was fine)? How otherwise you would know that the provided string is not parsable?

**Comment by ghost** [comment]
Please see https://github.com/vlang/v/pull/6550

**Comment by JalonSolov** [comment]
This was discussed a while ago on discord.  Basically, default functionality should be the same as the same-named function in other languages, in that almost all of them parse what they can and return that without including anything extra.  This is to avoid confusion for anyone expecting that same behavior from the same-named routine.

If you care to create a different routine that does return extra information such as this, that should be fine.

**Comment by JalonSolov** [comment]
For example, every language I have ever used that has a function named 'atoi' handles things exactly the same way:  If you pass it a string with no digits at the beginning, it returns 0, with no other info.  If you pass it a string with leading digits followed by non-digits, it returns the value of the digits and not other info, like this:
```
x := atoi('12somestring')
```
The result is that `x == 12`.
